benchmarks/high_volume_reads_test.go

```diff
@@ -17,126 +18,100 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-var _ = Describe("Scenario: High Volume Reads", func() {
-
-	scenarioCfgs := benchCfg.Scenarios.HighVolumeReads
-
-	samplingCfg := scenarioCfgs.Samples.SamplingConfiguration()
-	samplingRange := model.Duration(scenarioCfgs.Samples.Interval)
+var _ = Describe("High Volume Reads", func() {
+	var highVolumeReadsTest *config.HighVolumeReads
+	var generatorDpl client.Object
+	var querierDpls []client.Object
+	var samplingCfg gmeasure.SamplingConfig
+	var samplingRange model.Duration
 
 	BeforeEach(func() {
-		if !scenarioCfgs.Enabled {
-			Skip("High Volumes Reads Benchmark not enabled!")
+		if !benchCfg.Scenarios.IsReadTestRunnable() {
+			Skip("High Volumes Reads Benchmark not enabled")
 		}
+		highVolumeReadsTest = benchCfg.Scenarios.HighVolumeReads
+	})
 
-		generatorDpl := loadclient.CreateGenerator(scenarioCfgs.Generator, benchCfg.Generator)
+	Describe("Querying logs from Loki service", func() {
+		BeforeEach(func() {
+			querierDpls = querier.CreateQueriers(highVolumeReadsTest.Readers, benchCfg.Querier)
+			generatorDpl = loadclient.CreateGenerator(highVolumeReadsTest.LogGenerator(), benchCfg.Generator)
 
-		err := k8sClient.Create(context.TODO(), generatorDpl, &client.CreateOptions{})
-		Expect(err).Should(Succeed(), "Failed to deploy logger")
+			err := k8sClient.Create(context.TODO(), generatorDpl, &client.CreateOptions{})
+			Expect(err).Should(Succeed(), "Failed to deploy logger")
 
-		err = utils.WaitForReadyDeployment(k8sClient, generatorDpl, defaultRetry, defaultTimeout)
-		Expect(err).Should(Succeed(), "Failed to wait for ready logger deployment")
+			err = utils.WaitForReadyDeployment(k8sClient, generatorDpl, defaultRetry, defaultTimeout)
+			Expect(err).Should(Succeed(), "Failed to wait for ready logger deployment")
 
-		err = utils.WaitUntilReceivedBytes(metricsClient, scenarioCfgs.StartThreshold, defaultRange, defaultRetry, defaultTimeout)
-		Expect(err).Should(Succeed(), "Failed to wait until latch activated")
+			// Begin loading data into the Loki service so there is something to query for.
+			time.Sleep(time.Minute * 5)
```

Comment:
Shouldn't this be part of the scenario configuration to allow tuning? And btw why do we change the behavior here again from waiting for received bytes to time-based? I assume because are suite is short-lived but we generate massively the amount of data, e.g. for a day?

Reply:
I've found that the waiting for received bytes can be flaky. The goal is of the read path is not to stress the ingester with heavy write pressure. This makes the wait X time for X amount of data somewhat imprecise. In my opinion, it's about as "accurate" as just waiting X time, which is more easily configurable.

Someone could change this to generate a mass amount of data to simulate a "real" scenario, but I wouldn't call that the "default" for the test.